upb/def.c

```diff
@@ -898,11 +929,16 @@ const upb_msgdef *upb_symtab_lookupmsg(const upb_symtab *s, const char *sym) {
       unpack_def(v, UPB_DEFTYPE_MSG) : NULL;
 }
 
+static const void *symtab_lookup2(const upb_symtab *s, const char *sym,
+                                  size_t size, upb_deftype_t type) {
+  upb_value v;
+  return upb_strtable_lookup2(&s->syms, sym, size, &v) ? unpack_def(v, type)
```

Comment:
I understand that this is equivalent to an if, but relying on side effects inside ternaries is weird to me.  Can you write out the if?

Reply:
Ditto here: happy to move in this direction as a follow-up, but the next 5 or so functions are currently written with this pattern, so I'd prefer to keep as-is for now.

I can start a list of changes like this that we can aim to land en masse once the current set of work on my development branch has landed (to avoid a lot of spurious merge conflicts).